src/Aspire.Hosting/ApplicationModel/CustomResourceSnapshot.cs

```diff
@@ -71,12 +74,25 @@ public sealed record CustomResourceSnapshot
     /// The commands available in the dashboard for this resource.
     /// </summary>
     public ImmutableArray<ResourceCommandSnapshot> Commands { get; init; } = [];
+
+    /// <summary>
+    /// Gets the set of resource names that this resource will wait for. May be empty.
+    /// </summary>
+    public ImmutableArray<WaitForSnapshot> WaitFors { get; init; } = [];
+
+    /// <summary>
+    /// Gets the aggregate health status of the resource.
+    /// </summary>
+    /// <remarks>
+    /// This property's value is derived from <see cref="HealthReports"/> and is provided for convenience.
+    /// </remarks>
+    public HealthStatus HealthStatus => HealthReports.MinBy(r => r.Status)?.Status ?? HealthStatus.Healthy;
```

Comment:
This should be a cheap check. Let's just store the aggregate and not make this computed.